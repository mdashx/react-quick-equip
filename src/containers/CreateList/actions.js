import {
  ADD_USER_TO_LIST,
  UPDATE_USERNAME,
  DNE,
} from './constants';

export function updateUsername(username) {
  return {
    type: UPDATE_USERNAME,
    username,
  }
}

export function addUser() {
  return {
    type: ADD_USER_TO_LIST,
  }
}
