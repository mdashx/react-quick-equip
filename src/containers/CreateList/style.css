.myHeading {
    composes: headerFont from "styles/typography.css";
    composes: primary from "styles/colors.css";
}
