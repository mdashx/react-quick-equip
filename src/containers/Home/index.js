import React from 'react';

const Home = () => (
  <div>
    <h1>This is the home screen.</h1>
  </div>
);

export default Home;
