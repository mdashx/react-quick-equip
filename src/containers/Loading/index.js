import React from 'react';

class Loading extends React.Component {
  render() {
    return (
      <h2>Loading...</h2>
    );
  }
}

export default Loading;
