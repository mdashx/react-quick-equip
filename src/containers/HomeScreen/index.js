/* 
 *
 * Home Screen container (Redux connected component)
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { $selectorName } from 'features/$featureName/selectors';
import { $actionName } from 'features/$featureName/actions';

export class HomeScreen extends React.Component {
  render() {
    return (
      <div>
      </div>
    );
  }
}



const mapStateToProps = createStructuredSelector({
  $propName: $selectorName(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    $propName2: () => dispatch(--actionName()),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
