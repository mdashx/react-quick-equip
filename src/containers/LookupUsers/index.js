import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import ShowUserList from 'components/ShowUserList';
import ShowRepos from 'components/ShowRepos';
import { makeSelectUsernames } from 'features/findUser/selectors';
import { makeMostRecentRepos, makeMostRecentUser } from 'features/foundUser/selectors';
import { lookupUsername } from 'features/foundUser/actions';


class LookupUsers extends React.Component {
  render() {
    return (
      <div>
        <ShowUserList
            title="Lookup Users"
            usernames={this.props.usernames}
            clickHandler={this.props.lookupUsername}
        />
        <hr />
        <ShowRepos repos={this.props.mostRecentRepos} />
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  usernames: makeSelectUsernames(),
  mostRecentRepos: makeMostRecentRepos(),
  mostRecentUser: makeMostRecentUser(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    lookupUsername: (e) => { dispatch(lookupUsername(e.target.getAttribute('data-username'))) },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LookupUsers);
