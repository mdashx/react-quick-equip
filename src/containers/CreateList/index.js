import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import UserList from 'components/UserList';
import { addUser, updateUsername } from 'features/findUser/actions';
import {
  makeSelectUsernames,
  makeSelectCurrentUsername,
} from 'features/findUser/selectors';

import style from './style.css';

export class CreateList extends React.Component {
  render() {
    return (
      <div>
        <h1 className={style.myHeading}>
          What Github user's do you want to lookup?
        </h1>
        <form>
          <input
              value={this.props.currentUsername}
              onChange={this.props.updateUsername}
          />
          <span>&nbsp;</span>
          <input
              type="submit"
              value="Add User"
              onClick={this.props.addUser}
              data-username={this.props.currentUsername}
          />
        </form>
        <p>Click a username to delete it from the list.</p>
        <ul>
          <UserList
              usernames={this.props.usernames}
              clickHandler={this.props.removeUser}
          />
        </ul>
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  usernames: makeSelectUsernames(),
  currentUsername: makeSelectCurrentUsername(),
});

const mapDispatchToProps = dispatch => {
  return {
    updateUsername: e => dispatch(updateUsername(e.target.value)),
    addUser: e => {
      e.preventDefault();
      dispatch(addUser(e.target.getAttribute('data-username')));
    },
    removeUser: e => {
      console.log(`Remove user: ${e.target.getAttribute('data-username')}`);
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateList);
