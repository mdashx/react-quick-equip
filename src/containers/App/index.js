import React from 'react';
import Loadable from 'react-loadable';
import { Link, Route, Switch } from 'react-router-dom';

import Loading from 'containers/Loading';

const Home = Loadable({
  loader: () => import('containers/Home'),
  loading: Loading,
});

const CreateList = Loadable({
  loader: () => import('containers/CreateList'),
  loading: Loading,
});

const LookupUsers = Loadable({
  loader: () => import('containers/LookupUsers'),
  loading: Loading,
});

const SearchHistory = Loadable({
  loader: () => import('containers/SearchHistory'),
  loading: Loading,
});

const App = props => {
  return (
    <div>
      <div>
        <div id="navbar">
          <Link to="/">Home</Link>
          <span> -- </span>
          <Link to="/create">Create List Of Users To Lookup</Link>
          <span> -- </span>
          <Link to="/lookup">Lookup Users</Link>
          <span> -- </span>
          <Link to="/history">View Search History</Link>
        </div>
      </div>
      <div id="content">
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/create" component={CreateList} />
          <Route path="/lookup" component={LookupUsers} />
          <Route path="/history" component={SearchHistory} />
        </Switch>
      </div>
    </div>
  );
};

export default App;
