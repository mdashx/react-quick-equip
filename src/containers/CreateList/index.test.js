import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { CreateList } from 'containers/CreateList';

test('Renders...', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CreateList />, div);
});

test('Renders correctly', () => {
  const component = renderer.create(<CreateList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
