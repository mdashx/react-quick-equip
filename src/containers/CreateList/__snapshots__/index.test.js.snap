// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <h1>
    What Github user's do you want to lookup?
  </h1>
  <form>
    <input
      onChange={undefined}
      value={undefined}
    />
    <span>
      Â 
    </span>
    <input
      data-username={undefined}
      onClick={undefined}
      type="submit"
      value="Add User"
    />
  </form>
  <p>
    Click a username to delete it from the list.
  </p>
  <ul>
    <span />
  </ul>
</div>
`;
